packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -923,4 +881,77 @@ export default class Editor {
     }
 
     //#endregion
+
+    //#region Dark mode APIs
+
+    /**
+     * Set the dark mode state and transforms the content to match the new state.
+     * @param nextDarkMode The next status of dark mode. True if the editor should be in dark mode, false if not.
+     */
+    public setDarkModeState(nextDarkMode?: boolean) {
+        if (this.isDarkMode() == nextDarkMode) {
+            return;
+        }
+
+        const currentContent = this.getContent(
+            undefined /* triggerContentChangedEvent */,
+            true /* getSelectionMarker */);
+        this.core.inDarkMode = nextDarkMode;
+        if (nextDarkMode) {
+            this.setContent(
+                currentContent,
+                undefined /* triggerContentChangedEvent */,
+                true /* convertToDarkMode */);
+        } else {
+            this.setContent(currentContent);
+        }
+    }
+
+    /**
+     * Check if the editor is in dark mode
+     * @returns True if the editor is in dark mode, otherwise false
+     */
+    public isDarkMode(): boolean {
+        return this.core.inDarkMode;
+    }
+
+    /**
+     * Returns the dark mode options set on the editor
+     * @returns A DarkModeOptions object
+     */
+    public getDarkModeOptions(): DarkModeOptions {
+        return this.core.darkModeOptions;
+    }
+
+    /**
+     * Converter for dark mode that runs all child elements of a node through the content transform function.
+     * @param node The node containing HTML elements to convert.
+     */
+    private convertContentToDarkMode(node: Node): () => void {
+        let childElements: HTMLElement[] = [];
+
+        // Get a list of all the decendents of a node.
+        // querySelectorAll doesn't return a live list when called on an HTMLElement
+        // So we use getElementsByTagName instead for HTMLElement types.
+        if (node instanceof HTMLElement) {
+            childElements = Array.prototype.slice.call(node.getElementsByTagName('*'));
+            childElements.unshift(node);
+        } else if (node instanceof DocumentFragment) {
+            childElements = Array.prototype.slice.call(node.querySelectorAll('*'));
+        }
+
+        const darkModeOptions = this.getDarkModeOptions();

```

Comment:
This can only be pulled out to this level if the darkmode options will never change, since the lifetime of the lambda might outlive the lifetime of the options we get here otherwise. is that the case?

Reply:
You're right about that, but there's no mechanism to currently update the darkModeOptions. However, we might make those options mutable in the future (though I'm not sure that's good design). I think it's safer to put it inside the callback but outside the for-each